Refactoring

orizonJust opened the pull request:
Добавил удаление рецептов

OrizonJust committed:
Refactoring
Добавил удаление рецептов

Shveed commented on src/main/java/com/miti/server/model/entity/IngredientContext.java:
```diff
@@ -8,27 +8,27 @@
 @Data
-@JsonIgnoreProperties(ignoreUnknown = true, value = {"ingredients", "recipeIngredients"})
+@JsonIgnoreProperties(ignoreUnknown = true, value = {"recipeIngredients"})
 public class IngredientContext {
```
Давай для начала поменяем на ContextIngredient - так правильнее

Shveed commented on src/main/java/com/miti/server/controller/IngredientContextController.java:
```diff
@@ -14,7 +15,7 @@
 
   @PostMapping("/addIngredientContext")
-  public IngredientContext addIngredientContext(@RequestBody IngredientContext ingredientContext) {
+  public IngredientContext addIngredientContext(@RequestBody IngredientContextDTO ingredientContext) {
```
Надо избавиться от DTO. Не забудь ещё метод PUT протестить.

Shveed commented on src/main/java/com/miti/server/service/impl/IngredientContextServiceImpl.java:
```diff
@@ -22,19 +23,19 @@
   @Override
-  public IngredientContext addIngredientContext(IngredientContext ingredientContext) {
+  public IngredientContext addIngredientContext(IngredientContextDTO ingredientContext) {
     if (checkFields(
```
Раз уж ты решил оставить эти проверки, делай более подробное название. Не каждый сразу поймёт что ты именно чекаешь в этих полях

OrizonJust committed:
Убрал никому ненужную ДТО (в рот она колотись)
***
Разобрался со сраным JSON'ом (убрал jsonIgnore для обязательных полей, а для необязательных его добавил)
***
Добавил удаление с лица земли рецепта (сначала удаляются все связи, а после уже сам рецепт)

OrizonJust committed:
Изменил название класса IngredientContext на ContextIngredient, исправил все ошибки, которые всплыли после изменения.
***
Прописал все названия таблиц капсом, а также подписал все колонки капсом.(для лучшей читабельности)

OrizonJust committed:
Переименовал методы для проверки полей existsBy***

Shveed commented on src/main/java/com/miti/server/controller/IngredientContextController.java:
```diff
@@ -11,51 +11,51 @@
-  public List<IngredientContext> getAllIngredientContexts() {
-    return ingredientContextService.getAllIngredientContexts();
+  @GetMapping("/getAllContextIngredients")
+  public List<ContextIngredient> getAllIngredientContexts() {
```
Ну логично что методы тоже поменяются)

Shveed commented on src/main/java/com/miti/server/controller/IngredientContextController.java:
```diff
@@ -11,51 +11,51 @@
-  public List<IngredientContext> getIngredientContextByAmountLessThan(@RequestParam Integer amount) {
-    return ingredientContextService.getIngredientContextsByAmountLessThan(amount);
+  @GetMapping("/getContextIngredientsByAmountLessThan")
+  public List<ContextIngredient> getIngredientContextByAmountLessThan(@RequestParam Long amount) {
```
АНАЛогично в остальных

OrizonJust committed:
Полностью переработано удаление пользователей (удаление Comment + ContextIngredient -> удаление Recipe -> удаление UserComment -> удаление User)
Изменены названия методов contextIngredient

OrizonJust committed:
AppInit изменил addIngredientContext на addContextIngredient
